### PR TITLE
Rough implementation for scss dynamic resolver

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

File: src/services/cssNavigation.ts [hunk 1]
```diff
@@ -104,24 +126,85 @@ export class CSSNavigation {
...
+				originalBasename: uriFragments[uriFragments.length - 1],
+				normalizedUri,
+				normalizedBasename,
+				withBasename(newBaseName: string) {
+					return [...uriFragments.slice(0, -1), newBaseName].join('/');
+				}
+			};
+		}
+		
+		async function fileExists(p: string) {
```
Comment (aeschli) [inline src/services/cssNavigation.ts]: Use vcode-URI, uri to file path is not trivial

File: src/cssLanguageService.ts [hunk 2]
```diff
@@ -89,15 +97,39 @@ function handleCustomData(options?: LanguageServiceOptions) {
 
 export function getCSSLanguageService(options?: LanguageServiceOptions): LanguageService {
 	handleCustomData(options);
-	return createFacade(new Parser(), new CSSCompletion(), new CSSHover(), new CSSNavigation(), new CSSCodeActions(), new CSSValidation());
+	return createFacade(
+		new Parser(),
+		new CSSCompletion(),
+		new CSSHover(),
+		new CSSNavigation(),
```
Comment (aeschli) [inline src/cssLanguageService.ts]: I would pass in the fileSystemProvider in the constructor
- no need to add `fsProvider` to the facade

**Comment by octref** [comment]
This is ready for review.

**Comment by octref** [comment]
`findDocumentLinks` now return a promise of `DocumentLink[]` so we might need to publish new major version. Another choice is to instead introduce a `findDynamicDocumentLinks` method so we don't break existing API.

**Comment by aeschli** [comment]
Yes, could do findDocumentLinks2 so it's clear it's the replacement of the old one

**Pull request merged**